lib/selections/index.js

```diff
@@ -0,0 +1,904 @@
+   */
+  self.isCellSelected = (cell) => {
+    return isCellSelected(self.selections, cell.rowIndex, cell.viewColumnIndex);
+  };
+
+  /**
+   * Returns true if the selected columnIndex is selected on every row.
+   * @memberof canvasDatagrid
+   * @name isColumnSelected
+   * @method
+   * @param {number} columnIndex The column index to check.
+   */
+  self.isColumnSelected = (viewColumnIndex) => {
+    if (typeof viewColumnIndex !== 'number') return false;
+    return isColumnSelected(self.selections, viewColumnIndex);
+  };
+
+  /**
+   * Returns true if the selected rowIndex is selected on every column.
+   * @memberof canvasDatagrid
+   * @name isRowSelected
+   * @method
+   * @param {number} rowIndex The row index to check.
+   */
+  self.isRowSelected = (rowIndex) => {
+    if (typeof rowIndex !== 'number') return false;
+    return isRowSelected(self.selections, rowIndex);
+  };
+
+  self.clearSelections = (triggerEvent) => {
+    self.selections = [];
+    if (triggerEvent) self.dispatchSelectionChangedEvent();
+  };
+
+  /**
+   * Removes the selection.
+   * @memberof canvasDatagrid
+   * @name selectNone
+   * @param {boolean} dontDraw Suppress the draw method after the selection change.
+   * @method
+   */
+  self.selectNone = (dontDraw) => {
+    self.clearSelections(true);
+    if (dontDraw) return;
+    self.draw();
+  };
+
+  /**
+   * Selects every visible cell.
+   * @memberof canvasDatagrid
+   * @name selectAll
+   * @param {boolean} dontDraw Suppress the draw method after the selection change.
+   * @method
+   */
+  self.selectAll = function (dontDraw) {
+    const changed = addIntoSelections(self.selections, {
+      type: SelectionType.Columns,
+      startColumn: 0,
+      endColumn: self.getSchema().length - 1,
+    });
+    if (!changed || dontDraw) return;
+    self.draw();
+  };
+
+  /**
+   * @param {object} cell This method needs two properties in this parameter: `rowIndex`
+   * and `viewColumnIndex`
+   * @param {boolean} [supressEvent]
+   */
+  self.unselectCell = (cell, supressEvent) => {
+    const result = removeFromSelections(
+      self.selections,
+      normalizeSelection({
+        type: SelectionType.Cells,
+        startRow: cell.rowIndex,
+        startColumn: cell.viewColumnIndex,
+      }),
+    );
+    if (result && !supressEvent) self.dispatchSelectionChangedEvent();
+    return result;
+  };
+
+  /**
+   * @param {number} startRow rowViewIndex
+   * @param {number} endRow rowViewIndex
+   * @param {boolean} supressEvent When true, prevents the `selectionchanged` event from firing.
+   * @returns {boolean} It returns whether the selection changed
+   */
+  self.unselectRows = (startRow, endRow, supressEvent) => {
+    const result = removeFromSelections(
+      self.selections,
+      normalizeSelection({
+        type: SelectionType.Rows,
+        startRow: startRow,
+        endRow: endRow,
+      }),
+    );
+    if (result && !supressEvent) self.dispatchSelectionChangedEvent();
```

Comment:
Minor nit-pick: there is a typo in `supressEvent`; it should be `suppressEvent` (two p-s)